src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2013,15 +2041,28 @@ GenerateTaskMoveDependencyList(int64 colocationId,
 
 /*
  * UpdateShardMoveDependencies function updates the dependency maps with the latest move's taskId.
- * Currently there is just the nodeDependencies map to update. In the future we may add more.
  */
 static void
-UpdateShardMoveDependencies(uint64 colocationId, int64 taskId,
+UpdateShardMoveDependencies(PlacementUpdateEvent *move, uint64 colocationId, int64 taskId,
 							ShardMoveDependencies shardMoveDependencies)
 {
 	ShardMoveDependencyInfo *shardMoveDependencyInfo = hash_search(
 		shardMoveDependencies.colocationDependencies, &colocationId, HASH_ENTER, NULL);
 	shardMoveDependencyInfo->taskId = taskId;
+
+	bool found;
+	ShardMoveSourceNodeHashEntry *shardMoveSourceNodeHashEntry = hash_search(
+		shardMoveDependencies.nodeDependencies, &move->sourceNode->nodeId, HASH_ENTER,
+		&found);
+	if (found)
+	{
+		shardMoveSourceNodeHashEntry->taskIds = lappend_int(
+			shardMoveSourceNodeHashEntry->taskIds, taskId);
+	}
+	else
+	{
+		shardMoveSourceNodeHashEntry->taskIds = list_make1_int(taskId);
```

Comment:
It's quite annoying but int lists store int32s not int64s, so we cannot use them for taskIds. Best we can do afaik is to `palloc0(sizeof(int64))` and store that pointer in a pointer list.

Reply:
Good point